packages/uicore/src/components/ExpressionDropdown/ExpressionDropdown.tsx

```diff
@@ -159,18 +55,112 @@ export const NewExpressionDropdown = (props: NewExpressionDropdownProps): JSX.El
         currentNode = undefined
       }
     })
-
-    setOverflowListItems(listItems)
-  }, [query])
+    return listItems
+  }, [rootTrieNode, query])
 
   const dropDownItemClickHandler = (value: string): void => {
     const valueBefore = query.substring(0, query.lastIndexOf('<'))
     setQueryValue(`${valueBefore}<+${value}`)
   }
 
-  const visibleItemRenderer = getVisibleItemRenderer({ dropDownItemClickHandler, isOpen, setIsOpen })
+  const itemClickHandler = React.useCallback(
+    (valueTillHere: string, isOverflow = false): void => {
+      setOpenTargetElement(isOpen, setIsOpen, valueTillHere, true)
+      isOverflow && setIsPopoverOpen(prevState => !prevState)
+    },
+    [isOpen, setIsOpen]
+  )
+
+  const shallShowNesting = (item: TrieNode) => item.children.some(child => child.children.length !== 0)
```

Comment:
This is not added now, but can we rename `shallShowNesting` to `showDropdownIcon` or `hasGrandchildren`?